src/pages/iou/request/step/IOURequestStepScan/index.native.tsx

```diff
@@ -90,21 +91,21 @@ function IOURequestStepScan({
 
     const {translate} = useLocalize();
 
-    const askForPermissions = () => {
+    const askForPermissions = useCallback(() => {
         // There's no way we can check for the BLOCKED status without requesting the permission first
         // https://github.com/zoontek/react-native-permissions/blob/a836e114ce3a180b2b23916292c79841a267d828/README.md?plain=1#L670
         CameraPermission.requestCameraPermission?.()
             .then((status: string) => {
                 setCameraPermissionStatus(status);
 
                 if (status === RESULTS.BLOCKED) {
-                    FileUtils.showCameraPermissionsAlert();
+                    FileUtils.showCameraPermissionsAlert({screenName: SCREENS.RIGHT_MODAL.MONEY_REQUEST, iouType});
```

Comment:
@ravindra-encoresky I think we should save the current route in Onyx (using Navigation.getActiveRoute function)

Reply:
yes, you are correct. I am making necessary changes

Reply:
@DylanDylann i have made the necessary changes to use activeRoute and it is working as expected